runtime/sema/check_path_expression.go

```diff
@@ -19,40 +19,59 @@
 package sema
 
 import (
+	"regexp"
+
 	"github.com/onflow/cadence/runtime/ast"
 	"github.com/onflow/cadence/runtime/common"
 )
 
 func (checker *Checker) VisitPathExpression(expression *ast.PathExpression) ast.Repr {
 
-	ty, err := CheckPathLiteral(expression)
-	checker.report(err)
+	ty := CheckPathLiteral(
+		expression.Domain.Identifier,
+		expression.Identifier.Identifier,
+		func(errFunc func(err *ast.PathExpression) error) {
+			checker.report(errFunc(expression))
+		},
+	)
 
 	return ty
 }
 
-func CheckPathLiteral(expression *ast.PathExpression) (Type, error) {
+func CheckPathLiteral(domainString, identifier string, report func(func(err *ast.PathExpression) error)) Type {
 
 	// Check that the domain is valid
-
-	domainIdentifier := expression.Domain
-
-	domain, ok := common.AllPathDomainsByIdentifier[domainIdentifier.Identifier]
+	domain, ok := common.AllPathDomainsByIdentifier[domainString]
 	if !ok {
-		return PathType, &InvalidPathDomainError{
-			ActualDomain: domainIdentifier.Identifier,
-			Range:        ast.NewRangeFromPositioned(domainIdentifier),
-		}
+		report(func(expression *ast.PathExpression) error {
+			return &InvalidPathDomainError{
+				ActualDomain: domainString,
+				Range:        ast.NewRangeFromPositioned(expression.Domain),
+			}
+		})
+		return PathType
+	}
+
+	// Check that the identifier is valid
+	isValidIdentifier := regexp.MustCompile(`^[A-Za-z_][A-Za-z0-9_]*$`).MatchString
```

Comment:
This re-compiles the regular expression on each call of `CheckPathLiteral`. 
Refactor the regular expression out into a global